pkg/azure/access/helpers/disk.go

```diff
@@ -32,7 +31,7 @@ const (
 // DeleteDisk deletes disk for passed in resourceGroup and diskName.
 // NOTE: All calls to this Azure API are instrumented as prometheus metric.
 func DeleteDisk(ctx context.Context, client *armcompute.DisksClient, resourceGroup, diskName string) (err error) {
-	defer instrument.RecordAzAPIMetric(err, diskDeleteServiceLabel, time.Now())
+	defer instrument.APIMetricRecorderFn(diskDeleteServiceLabel, &err)
```

Comment:
```suggestion
	defer instrument.APIMetricRecorderFn(diskDeleteServiceLabel, &err)()
```

We need to execute the returned function to record the metric. Same change at other places as well

Reply:
Thanks for pointing this out, changes are made and unit tests are now added which were missing before. PTAL